docs/Setting-up-monorepo.md

```diff
@@ -0,0 +1,356 @@
+
+```
+touch packages/form-validator/package.json
+```
+
+7. Add package declaration to the file:
+
+```
+{
+  "name": "form-validator",
+  "version": "1.0.0"
+}
+```
+
+## Step 6: Add the form validator utility to the mobile app
+
+1. Add the form validator utility to the app's `package.json` file:
+
+`apps/mobile/package.json`
+
+```
+"expo-status-bar": "~1.12.1",
+"form-validator": "workspace:^", // <- Add this new line
+"i18n-js": "3.9.2",
+```
+
+2. Import the `isEmailValid` function:
+
+At the top of the `LoginScreen.tsx` file, add the import statement for the isEmailValid function. Ensure that the path corresponds to where your form-validator utility is located within your monorepo.
+
+`apps/mobile/app/screens/LoginScreen.tsx`
+
+```
+import { AppStackScreenProps } from "../navigators"
+import { colors, spacing } from "../theme"
+import { isEmailValid } from "form-validator" // <- Add this new line
+```
+
+3. Add new `Text` component that will display if the entered email is valid or not.
+
+`apps/mobile/app/screens/LoginScreen.tsx`
+
+```
+<Text tx="loginScreen.enterDetails" preset="subheading" style={$enterDetails} />
+      {attemptsCount > 2 && <Text tx="loginScreen.hint" size="sm" weight="light" style={$hint} />}
+
+/** Add these next 5 lines **/
+<Text
+  text={isEmailValid(authEmail) ? "It is a valid email" : "It is not a valid email"}
+  preset="subheading"
+  style={$enterDetails}
+/>
+```
+
+## Step 7: Add the form validator utility to the web app
+
+1. Add the form validator utility to the app's `package.json` file:
+
+`apps/web/package.json`
+
+```
+"dependencies": {
+  "form-validator": "workspace:^"
+}
+```
+
+2. Create an `index.js` file to handle the form submission and validation:
+
+```
+// Import the validator package
+import { isEmailValid } from 'form-validator';
+
+// Handle form submission
+document.getElementById('signup-form').addEventListener('submit', function(event) {
+  event.preventDefault(); // Prevent the form from submitting the default way
+
+  const email = document.getElementById('email').value;
+  const password = document.getElementById('password').value;
+
+  // Validate email
+  if (!isEmailValid(email)) {
+    alert('Please enter a valid email address.');
+    return;
+  }
+
+  // Simple password validation (e.g., check if not empty)
+  if (password.length === 0) {
+    alert('Please enter a password.');
+    return;
+  }
+
+  // If both are valid, submit the form (or do something else)
+  alert('Form submitted successfully!');
+});
+```
+
+## Step 8: Run both apps to make sure logic was added
+
+1. Make sure dependencies are installed.
+
+```
+yarn install
+```
+
+2. Run React Native app (make sure you have your [environment setup](https://reactnative.dev/docs/set-up-your-environment)).
+
+For iOS:
+
+```
+cd apps/mobile
+yarn ios
+```
+
+For Android:
+
+```
+cd apps/mobile
+yarn android
+```
+
+3. Run web app
+
+```
+cd apps/web
+yarn start
+```
```

Comment:
We should probably address live reloading across web and mobile, when not to share logic / code, bundling considerations, and more. We should also make sure this guide, if appropriate, conforms to @trevor-coleman's ideas in [this internal wiki](https://infinitered.slab.com/posts/sharing-config-in-monorepos-u2av3f61) (open to disagreement with how he presented it though, use discernment please.)

Reply:
Adding a shared eslint config package or a shared tsconfig package would be a great addition to this guide, as that's the place where most people screw up.